node/src/main/java/io/airlift/node/NodeInfo.java

```diff
@@ -148,6 +155,9 @@ public NodeInfo(String environment,
                 throw new UncheckedIOException(e);
             }
         }
+        else if (annotationString != null) {
+            this.annotations = ImmutableMap.copyOf(replaceEnvironmentVariables(Splitter.on(",").withKeyValueSeparator("=").split(annotationString)));
```

Comment:
Splitter should be moved to configuration class and used there. Field should be `Map<String, String>` instead of a `String`.

`replaceEnvironmentVariables` is redundant - it's called by the Bootstrap. Remove

Reply:
In that case reading and parsing the file should also be done by the config object for consistency.

Reply:
Configs seem to be designed with one input translating to one output. 

With that in mind it makes sense to change NodeInfo annotations input type to Map and move the file loading to the config file.